src/Heatmap/HeatMapInterpolationPoints.swift

```diff
@@ -155,7 +155,7 @@ public class HeatMapInterpolationPoints: NSObject {
         // Try to make as few clusters as possible; start with 1 and increment as needed
         var numClusters = 1
         
-        if (data.count > 0) {
+        if !data.isEmpty {
```

Comment:
Can you share the motivation for this? Any benchmark numbers to compare the speed improvement from this change? I would presume that the Swift compiler will optimize for the count operation on an array making the speed improvements of this change negligible. But if you can share specific data on it, happy to accept this change.

Reply:
In terms of performance, isEmpty executes in O(1) however count does it in O(N), where N is the amount of characters of the String.

![Screenshot 2021-07-19 at 9 43 27 PM](https://user-images.githubusercontent.com/37391704/126192464-bd8d2b76-be51-4124-9705-1550fbd751e2.png)

Reply:
Previously it was checking  whether `data` has value or not by using `if (data.count > 0) `,  Instead of this swift already provided `isEmpty` check which will be fast and improve the performance on scale.

Reply:
According to the link you shared, only collections that don't conform to [RandomAccessCollection](https://developer.apple.com/documentation/swift/randomaccesscollection) will iterate through all the elements in the collection when calling `count`. In the case of `HeatMapInterpolationPoints`, the underlying collection is an `Array` and so `count` access will be O(1). In the case of a String, however, calling `count` is not O(1) in which case `isEmpty` is certainly more performant as depicted in the screenshot you shared.

Given this line of reasoning, this change should be reverted since it's mostly preferential and not based on performance improvements.

Reply:
Thanks @arriolac for the information :),  reverted the changes back